### PR TITLE
add #[must_use] new_call function to show warning

### PR DESCRIPTION
**Pull request by mubarak23**

### CODE REVIEW

File: examples/brainfuck/src/main.rs [hunk 1]
```diff
@@ -107,6 +107,7 @@ fn codegen<'a, 'ctx>(ops: &[Op], context: &'a gccjit::Context<'ctx>) -> bool {
     let mut current_block = brainf_main.new_block("entry_block");
     // now we have to zero out the giant buffer we just allocated on the stack.
     let zero_access = context.new_array_access(None, array.to_rvalue(), context.new_rvalue_zero(int_ty));
+    // a call to context.new_call will give us warning
```
Comment (antoyo) [inline examples/brainfuck/src/main.rs:110]: I would only keep this comment, so please remove the others.

And please edit the comment to something like this:
```suggestion
    // A function call that is done for its side effects must be sent to add_eval.
```

**Review by antoyo — COMMENTED** [review]
A small nitpick.

File: examples/brainfuck/src/main.rs (hunk 1 above)
Comment (mubarak23) [reply examples/brainfuck/src/main.rs:110]: Alright

**Comment by mubarak23** [comment]
@antoyo these changes has been push as requested

File: examples/brainfuck/src/main.rs [hunk 2]
```diff
@@ -172,11 +173,13 @@ fn codegen<'a, 'ctx>(ops: &[Op], context: &'a gccjit::Context<'ctx>) -> bool {
             }
             Op::Input => {
                 let access = context.new_array_access(None, array.to_rvalue(), memory_ptr.to_rvalue());
+
```
Comment (antoyo) [inline examples/brainfuck/src/main.rs:176]: Please remove this new line and the others below.

**Review by antoyo — COMMENTED** [review]
Another nitpick :) .
I can merge after you fix it.
Thanks a lot!

File: examples/brainfuck/src/main.rs (hunk 2 above)
Comment (mubarak23) [reply examples/brainfuck/src/main.rs:176]: Alright

**Comment by mubarak23** [comment]
@antoyo new line remove

**Comment by antoyo** [comment]
> @antoyo new line remove

There are a few more that are left [here](https://github.com/rust-lang/gccjit.rs/pull/30/files#diff-0b2b94ab932650d132c3e0c5381fe2e74dfd28b6b3e794e7955ceba987a3b7e9R181) and [there](https://github.com/rust-lang/gccjit.rs/pull/30/files#diff-7cb0666cd0e7bcb77f8e3d77bbf60b59691dd977cb8842e7108ce9c6081eda4bR38), so please remove them as well.

**Comment by mubarak23** [comment]
oh, really,

**Comment by mubarak23** [comment]
@antoyo remove space

**Comment by antoyo** [comment]
Thanks for your contribution.

**Comment by antoyo** [comment]
@mubarak23: Also, please do not ping me as I get a notification from GitHub so I already know you pushed your changes :) .
I'll review them when I have time (still work hours here for me).

**Pull request merged**